Compilation error on aarch64-linux

<!---
Thank you for reporting a problem and suggesting improvements. Please provide the below information to make sure we can effectively deal with the issue reported. For the most precise version information, see the first line of console output or run with --version. Please attach or include example code that allows us to reproduce the problem.
--->

CBMC version: 5.72.1 (commit de504f532f407b51fcdb9ef7ec61b4bae4fde49e)
Operating system: Ubuntu 22.04
Exact command line resulting in the issue: `make`
What behaviour did you expect: It should compile
What happened instead: Compilation fails

Commands:

```shell
$ cmake .. -DWITH_JBMC=OFF -Dsat_impl=cadical
$ make
Checking /home/jiegec/nixpkgs-dev/cbmc/src/ansi-c/library/stdio.c
__libcheck.c: In function ‘vfscanf’:
__libcheck.c:986:52: error: incompatible type for argument 1 of ‘__CPROVER_POINTER_OFFSET’
  986 |   while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
      |                                                    ^~~
      |                                                    |
      |                                                    va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:62:44: note: expected ‘const void *’ but argument is of type ‘va_list’
   62 | __CPROVER_ssize_t __CPROVER_POINTER_OFFSET(const void *);
      |                                            ^~~~~~~~~~~~
__libcheck.c:987:31: error: incompatible type for argument 1 of ‘__CPROVER_OBJECT_SIZE’
  987 |         __CPROVER_OBJECT_SIZE(arg))
      |                               ^~~
      |                               |
      |                               va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:63:40: note: expected ‘const void *’ but argument is of type ‘va_list’
   63 | __CPROVER_size_t __CPROVER_OBJECT_SIZE(const void *);
      |                                        ^~~~~~~~~~~~
__libcheck.c: In function ‘__isoc99_vfscanf’:
__libcheck.c:1033:52: error: incompatible type for argument 1 of ‘__CPROVER_POINTER_OFFSET’
 1033 |   while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
      |                                                    ^~~
      |                                                    |
      |                                                    va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:62:44: note: expected ‘const void *’ but argument is of type ‘va_list’
   62 | __CPROVER_ssize_t __CPROVER_POINTER_OFFSET(const void *);
      |                                            ^~~~~~~~~~~~
__libcheck.c:1034:31: error: incompatible type for argument 1 of ‘__CPROVER_OBJECT_SIZE’
 1034 |         __CPROVER_OBJECT_SIZE(arg))
      |                               ^~~
      |                               |
      |                               va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:63:40: note: expected ‘const void *’ but argument is of type ‘va_list’
   63 | __CPROVER_size_t __CPROVER_OBJECT_SIZE(const void *);
      |                                        ^~~~~~~~~~~~
__libcheck.c: In function ‘vsscanf’:
__libcheck.c:1155:52: error: incompatible type for argument 1 of ‘__CPROVER_POINTER_OFFSET’
 1155 |   while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
      |                                                    ^~~
      |                                                    |
      |                                                    va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:62:44: note: expected ‘const void *’ but argument is of type ‘va_list’
   62 | __CPROVER_ssize_t __CPROVER_POINTER_OFFSET(const void *);
      |                                            ^~~~~~~~~~~~
__libcheck.c:1156:31: error: incompatible type for argument 1 of ‘__CPROVER_OBJECT_SIZE’
 1156 |         __CPROVER_OBJECT_SIZE(arg))
      |                               ^~~
      |                               |
      |                               va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:63:40: note: expected ‘const void *’ but argument is of type ‘va_list’
   63 | __CPROVER_size_t __CPROVER_OBJECT_SIZE(const void *);
      |                                        ^~~~~~~~~~~~
__libcheck.c: In function ‘__isoc99_vsscanf’:
__libcheck.c:1188:52: error: incompatible type for argument 1 of ‘__CPROVER_POINTER_OFFSET’
 1188 |   while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
      |                                                    ^~~
      |                                                    |
      |                                                    va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:62:44: note: expected ‘const void *’ but argument is of type ‘va_list’
   62 | __CPROVER_ssize_t __CPROVER_POINTER_OFFSET(const void *);
      |                                            ^~~~~~~~~~~~
__libcheck.c:1189:31: error: incompatible type for argument 1 of ‘__CPROVER_OBJECT_SIZE’
 1189 |         __CPROVER_OBJECT_SIZE(arg))
      |                               ^~~
      |                               |
      |                               va_list
In file included from ./library/cprover.h:43,
                 from <command-line>:
./library/../cprover_builtin_headers.h:63:40: note: expected ‘const void *’ but argument is of type ‘va_list’
   63 | __CPROVER_size_t __CPROVER_OBJECT_SIZE(const void *);
      |                                        ^~~~~~~~~~~~
rm: cannot remove '__libcheck.s': No such file or directory
make[2]: *** [src/ansi-c/CMakeFiles/ansi-c.dir/build.make:267: src/ansi-c/library-check.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:1858: src/ansi-c/CMakeFiles/ansi-c.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

--- a/src/ansi-c/library/stdio.c
+++ b/src/ansi-c/library/stdio.c
@@ -997,7 +997,7 @@ int vfscanf(FILE *restrict stream, const char *restrict format, va_list arg)
   }
 
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&arg) <
         __CPROVER_OBJECT_SIZE(arg))
   {
     void *a = va_arg(arg, void *);
@@ -1046,7 +1046,7 @@ __CPROVER_HIDE:;
   }
 
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&arg) <
         __CPROVER_OBJECT_SIZE(arg))
   {
     void *a = va_arg(arg, void *);
@@ -1095,7 +1095,7 @@ int __stdio_common_vfscanf(
   }
 
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(args) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&args) <
         __CPROVER_OBJECT_SIZE(args))
   {
     void *a = va_arg(args, void *);
@@ -1174,7 +1174,7 @@ __CPROVER_HIDE:;
   int result = __VERIFIER_nondet_int();
   (void)*s;
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&arg) <
         __CPROVER_OBJECT_SIZE(arg))
   {
     void *a = va_arg(arg, void *);
@@ -1209,7 +1209,7 @@ __CPROVER_HIDE:;
   int result = __VERIFIER_nondet_int();
   (void)*s;
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(arg) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&arg) <
         __CPROVER_OBJECT_SIZE(arg))
   {
     void *a = va_arg(arg, void *);
@@ -1250,7 +1250,7 @@ int __stdio_common_vsscanf(
 
   (void)*s;
   (void)*format;
-  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(args) <
+  while((__CPROVER_size_t)__CPROVER_POINTER_OFFSET(*(void **)&args) <
         __CPROVER_OBJECT_SIZE(args))
   {
     void *a = va_arg(args, void *);

